src/pages/auth/register.tsx

```diff
@@ -0,0 +1,182 @@
+import { motion } from 'framer-motion';
+import { useState } from 'react';
+
+import { Button, Input } from '@/components';
+import Card from '@/components/atoms/Card';
+import AuthLayout from '@/layouts/authLayout';
+
+import LinkButton from '../../components/atoms/LinkButton';
+
+const Register = () => {
+  const [username, setUserName] = useState('');
+  const [usernameInvalid, setUsernameInvalid] = useState(false);
+  const [email, setEmail] = useState('');
+  const [emailInvalid, setEmailInvalid] = useState(false);
+  const [passwordInvalid, setPasswordInvalid] = useState(false);
+  const [password, setPassword] = useState('');
+
+  const usernameChange = (e: any) => {
+    setUsernameInvalid(false);
+    setUserName(e.target.value);
+  };
+
+  const passwordChange = (e: any) => {
+    setPasswordInvalid(false);
+    setPassword(e.target.value);
+  };
+
+  const emailChange = (e: any) => {
+    setEmailInvalid(false);
+    setEmail(e.target.value);
+  };
+
+  const emailValidation = () => {
+    const regEx = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i;
+    if (regEx.test(email)) {
+      setEmailInvalid(false);
+    } else {
+      setEmailInvalid(true);
+    }
+  };
+
+  const passwordValidation = () => {
+    const regEx =
+      /^(?=.*[a-zA-Z])(?=.*\d)(?=.*[!@#$%^&*()_+])[A-Za-z\d][A-Za-z\d!@#$%^&*()_+]{5,20}$/;
+    if (regEx.test(password)) {
+      setPasswordInvalid(false);
+    } else {
+      setPasswordInvalid(true);
+    }
+  };
```

Comment:
Let's create a `/utils/auth.ts` folder containing all the helper functions we're using for auth